src/Symfony/Component/Uid/AbstractUid.php

```diff
@@ -37,6 +37,34 @@ abstract public static function isValid(string $uid): bool;
      */
     abstract public static function fromString(string $uid): self;
 
+    /**
+     * @return static
+     *
+     * @throws \InvalidArgumentException When the passed value is not valid
+     */
+    abstract public static function fromBinary(string $uid): self;
```

Comment:
wouldn't it be possible to provide generic implementations?

Reply:
I will have a look.